### ISSUE TITLE
migrate tool exception - trying to migrate pkg:stream_transform

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/sdk#44178]
```
Migrating /Users/kevmoo/github/stream_transform

See https://dart.dev/go/null-safety-migration for a migration guide.

Analyzing project...
[-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|]No analysis issues found.

Generating migration suggestions...
[--------------------------------------------------------------------------------------------------------------------------------------|                                                                                                                                       ]Aborting migration due to an exception.  This most likely is due to a
bug in the migration tool.  Please consider filing a bug report at:

https://github.com/dart-lang/sdk/issues/new
To attempt to perform migration anyway, you may re-run with
--ignore-exceptions.

Exception details:

RangeError (index): Invalid value: Valid value range is empty: 0

#0      List.[] (dart:core-patch/growable_array.dart:177:60)
#1      _AssignmentChecker._checkAssignment_recursion (package:nnbd_migration/src/edge_builder.dart:3429:51)
#2      _AssignmentChecker._checkAssignment (package:nnbd_migration/src/edge_builder.dart:3312:5)
#3      _AssignmentChecker._checkAssignment_recursion (package:nnbd_migration/src/edge_builder.dart:3427:9)
#4      _AssignmentChecker._checkAssignment (package:nnbd_migration/src/edge_builder.dart:3312:5)
#5      EdgeBuilder._handleAssignment (package:nnbd_migration/src/edge_builder.dart:2334:11)
#6      EdgeBuilder._handleInvocationArguments (package:nnbd_migration/src/edge_builder.dart:2859:7)
#7      EdgeBuilder.visitMethodInvocation (package:nnbd_migration/src/edge_builder.dart:1260:24)
#8      MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:7120:49)
#9      EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2151:22)
#10     EdgeBuilder._handleAssignment (package:nnbd_migration/src/edge_builder.dart:2265:20)
#11     EdgeBuilder.visitExpressionFunctionBody (package:nnbd_migration/src/edge_builder.dart:789:5)
#12     ExpressionFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:3493:15)
#13     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2151:22)
#14     EdgeBuilder._handleExecutableDeclaration (package:nnbd_migration/src/edge_builder.dart:2415:7)
#15     EdgeBuilder.visitMethodDeclaration (package:nnbd_migration/src/edge_builder.dart:1208:5)
#16     MethodDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:6967:49)
#17     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2151:22)
#18     EdgeBuilder._dispatchList (package:nnbd_migration/src/edge_builder.dart:2165:7)
#19     EdgeBuilder.visitClassOrMixinOrExtensionDeclaration (package:nnbd_migration/src/edge_builder.dart:603:7)
#20     EdgeBuilder.visitExtensionDeclaration (package:nnbd_migration/src/edge_builder.dart:803:5)
#21     ExtensionDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:3807:49)
#22     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2151:22)
#23     EdgeBuilder.visitNode (package:nnbd_migration/src/edge_builder.dart:1304:9)
#24     GeneralizingAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:202:51)
#25     CompletenessTracker.visitCompilationUnit.<anonymous closure> (package:nnbd_migration/src/utilities/completeness_tracker.dart:52:24)
#26     PermissiveModeVisitor.reportExceptionsIfPermissive (package:nnbd_migration/src/utilities/permissive_mode.dart:26:24)
#27     CompletenessTracker.visitCompilationUnit (package:nnbd_migration/src/utilities/completeness_tracker.dart:43:5)
#28     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2112:49)
#29     NullabilityMigrationImpl.processInput (package:nnbd_migration/src/nullability_migration_impl.dart:255:12)
#30     NonNullableFix.processUnit (package:nnbd_migration/src/front_end/non_nullable_fix.dart:167:15)
#31     _FixCodeProcessor.runLaterPhases.<anonymous closure> (package:nnbd_migration/migration_cli.dart:1091:19)
#32     _FixCodeProcessor.runLaterPhases.<anonymous closure> (package:nnbd_migration/migration_cli.dart:1089:28)
#33     _FixCodeProcessor.processResources (package:nnbd_migration/migration_cli.dart:1023:30)
<asynchronous suspension>
#34     _FixCodeProcessor.runLaterPhases (package:nnbd_migration/migration_cli.dart:1089:5)
<asynchronous suspension>
#35     MigrationCliRunner.run (package:nnbd_migration/migration_cli.dart:694:24)
<asynchronous suspension>
#36     MigrateCommand.run (package:nnbd_migration/migration_cli.dart:207:7)
<asynchronous suspension>
#37     CommandRunner.runCommand (package:args/command_runner.dart:197:13)
<asynchronous suspension>
#38     DartdevRunner.runCommand (package:dartdev/dartdev.dart:249:24)
<asynchronous suspension>
#39     runDartdev (package:dartdev/dartdev.dart:112:14)
<asynchronous suspension>
#40     main (file:///b/s/w/ir/cache/builder/sdk/pkg/dartdev/bin/dartdev.dart:11:3)
<asynchronous suspension>

```

**Comment by mit-mit** [comment]
@kevmoo which dart version?

**Comment by kevmoo** [comment]
`2.12.0-29.0.dev` – the "please consider filing a bug" message should include this – another feature request 😄

**Comment by mit-mit** [comment]
cc @stereotype441

**Comment by stereotype441** [comment]
> `2.12.0-29.0.dev` – the "please consider filing a bug" message should include this – another feature request 😄

@kevmoo Good idea.  https://dart-review.googlesource.com/c/sdk/+/172040

**Comment by stereotype441** [comment]
This was fixed in c0586277d78821e7c50c7e8aca507846e5844561.
